aws/resource_aws_elasticsearch_domain.go

```diff
@@ -642,6 +681,13 @@ func resourceAwsElasticSearchDomainRead(d *schema.ResourceData, meta interface{}
 		d.Set("log_publishing_options", m)
 	}
 
+	if ds.DomainEndpointOptions != nil {
+		m := make(map[string]interface{})
+		m["require_https"] = ds.DomainEndpointOptions.EnforceHTTPS
+		m["tls_security_policy"] = ds.DomainEndpointOptions.TLSSecurityPolicy
+		d.Set("domain_endpoint_options", m)
+	}
```

Comment:
Similar but opposite concept here with "flatten" functions 👍 Also, when using `d.Set()` with aggregate types (TypeList, TypeSet, TypeMap), [we should perform error checking to prevent issues where the code is not properly able to set the Terraform state](https://www.terraform.io/docs/extend/best-practices/detecting-drift.html#error-checking-aggregate-types).

```go
	if err := d.Set("domain_endpoint_options", flattenESDomainEndpointOptions(ds.DomainEndpointOptions)); err != nil {
		return fmt.Errorf("error setting domain_endpoint_options: %s", err)
	}

// ...

func flattenESDomainEndpointOptions(domainEndpointOptions *elasticsearch.DomainEndpointOptions) []interface{} {
	if domainEndpointOptions == nil {
		return nil
	}

	m := map[string]interface{}{
		"enforce_https":       aws.BoolValue(domainEndpointOptions.EnforceHTTPS),
		"tls_security_policy": aws.StringValue(domainEndpointOptions.TLSSecurityPolicy),
	}

	return []interface{}{m}
}
```